editoast/src/views/documents.rs

```diff
@@ -131,72 +134,91 @@ mod tests {
-    use crate::fixtures::tests::TestFixture;
-    use crate::views::tests::create_test_service;
-
-    #[rstest]
-    async fn get_document(
-        #[future] document_example: TestFixture<Document>,
-        db_pool: Data<DbConnectionPool>,
-    ) {
-        let service = create_test_service().await;
-        let doc = document_example.await;
-
-        let doc_key = doc.id();
-        let url = format!("/documents/{}", doc_key);
-
-        // Should succeed
-        let request = TestRequest::get().uri(&url).to_request();
-        let response = call_service(&service, request).await;
-        assert!(response.status().is_success());
-
-        // Delete the document
-        assert!(doc
-            .model
-            .delete(&mut db_pool.get().await.unwrap())
-            .await
-            .unwrap());
-
-        // Should fail
-        let request = TestRequest::get().uri(&url).to_request();
-        let response = call_service(&service, request).await;
-        assert!(response.status().is_client_error());
-    }
+    use crate::views::tests::create_test_connection_pool;
+    use crate::views::tests::create_test_service_with_connection_pool;
 
     #[derive(Deserialize, Clone, Debug)]
     struct PostDocumentResponse {
         document_key: i64,
     }
 
+    /// Create a document in the database using the connection pool in test mode
+    async fn create_document(connection_pool: &DbConnectionPoolV2) -> Document {
+        let mut conn = connection_pool
+            .get()
+            .await
+            .expect("Failed to get connection pool");
+        Document::changeset()
+            .data("Document post test data".as_bytes().to_vec())
+            .content_type(String::from("text/plain"))
+            .create(&mut conn)
+            .await
+            .expect("Failed to create document")
+    }
+
     #[rstest]
-    async fn document_post(db_pool: Data<DbConnectionPool>) {
-        let service = create_test_service().await;
+    async fn document_post() {
+        let pool = create_test_connection_pool().await;
+        let service = create_test_service_with_connection_pool(pool.clone()).await;
 
-        // Insert document
         let request = TestRequest::post()
             .uri("/documents")
             .insert_header(ContentType::plaintext())
-            .set_payload("Test data".as_bytes().to_vec())
+            .set_payload("Document post test data".as_bytes().to_vec())
             .to_request();
-        let response: PostDocumentResponse = call_and_read_body_json(&service, request).await;
-
-        // Delete the document
-        assert!(
-            Document::delete_static(&mut db_pool.get().await.unwrap(), response.document_key)
-                .await
-                .unwrap()
-        );
+
+        // Insert document
+        let create_response: PostDocumentResponse =
+            call_and_read_body_json(&service, request).await;
+
+        assert!(create_response.document_key > 0);
```

Comment:
You should try to retrieve the document back with the ID and check that the content matches.